pet-store-react-hooks/src/DeletePetModal/DeleteModal.test.tsx

```diff
@@ -0,0 +1,203 @@
+  expect(headerCloseButton).toBeInTheDocument();
+
+  // Pet info
+  const petName = await screen.findByText('Name: Kenny');
+  expect(petName).toBeInTheDocument();
+
+  const petKind = await screen.findByText('Kind: Parrot');
+  expect(petKind).toBeInTheDocument();
+
+  const petDateAdded = await screen.findByText('Date added: 27 Oct 2022');
+  expect(petDateAdded).toBeInTheDocument();
+
+  // Buttons
+  const deleteButton = await screen.findByRole('button', { name: 'Delete' });
+  expect(deleteButton).toBeInTheDocument();
+
+  const cancelButton = await screen.findByRole('button', { name: 'Cancel' });
+  expect(cancelButton).toBeInTheDocument();
+});
+
+test('Modal is closed on header close button click', async () => {
+  onClose.mockClear();
+  const user = userEvent.setup();
+
+  render(
+    <DeletePetModal
+      pet={petForDelete}
+      petKind={petKindsMap.get(petForDelete.kind)}
+      onClose={onClose}
+      onDeleted={onDeleted}
+    />
+  );
+
+  const headerCloseButton = await screen.findByRole('button', { name: 'X' });
+  await user.click(headerCloseButton);
+
+  expect(onClose).toHaveBeenCalled();
+});
+
+test('Modal is closed on body Cancel button click', async () => {
+  onClose.mockClear();
+  const user = userEvent.setup();
+
+  render(
+    <DeletePetModal
+      pet={petForDelete}
+      petKind={petKindsMap.get(petForDelete.kind)}
+      onClose={onClose}
+      onDeleted={onDeleted}
+    />
+  );
+
+  const cancelButton = await screen.findByRole('button', { name: 'Cancel' });
+  await user.click(cancelButton);
+
+  expect(onClose).toHaveBeenCalled();
+});
+
+test('Modal is closed on modal backdrop click', async () => {
+  onClose.mockClear();
+  const user = userEvent.setup();
+
+  render(
+    <DeletePetModal
+      pet={petForDelete}
+      petKind={petKindsMap.get(petForDelete.kind)}
+      onClose={onClose}
+      onDeleted={onDeleted}
+    />
+  );
+
+  const modalBackdrop = await screen.findByLabelText('modal backdrop');
+  await user.click(modalBackdrop);
+
+  expect(onClose).toHaveBeenCalled();
+});
+
+test('Modal is locked while deleting the pet and it is closed when deletion is successfull', async () => {
+  onClose.mockClear();
+  onDeleted.mockClear();
+
+  const user = userEvent.setup();
+  const waitHandle = new WaitHandle();
+
+  server.resetHandlers(
+    http.delete(`${apiBaseUrl}/pet/:petId`, async () => {
+      await waitHandle.wait();
+
+      const pet = pets.find((x) => x.petId === 44);
+      return HttpResponse.json(pet);
+    })
+  );
+
+  render(
+    <DeletePetModal
+      pet={petForDelete}
+      petKind={petKindsMap.get(petForDelete.kind)}
+      onClose={onClose}
+      onDeleted={onDeleted}
+    />
+  );
+
+  const deleteButton = await screen.findByRole('button', { name: 'Delete' });
+  await user.click(deleteButton);
+  expect(deleteButton).toBeDisabled();
+
+  // Question: Is there any point of testing the disabled state of the buttons here?
```

Comment:
You don't need to test that a disabled button is not calling it's onClick handler.